### PR TITLE
Feature #68 state time limit option

### LINKED ISSUE #68: Option for changing state time limit
It would sometimes be useful to change the value for state time limit from 3min to something else. For example when IdP requires user to fill some details on first login or when it requires user to read a license agreement, etc.

### PR DESCRIPTION
**Pull request by anttileppa**
Resolves #68

### CODE REVIEW

**Comment by daggerhart** [comment]
Thanks! I think this is a great idea.

I just have 2 very-very minor formatting requests. I'll comment on the code.

File: includes/openid-connect-generic-client.php [hunk 1]
```diff
@@ -25,15 +25,17 @@ class OpenID_Connect_Generic_Client {
 	 * @param $endpoint_userinfo
 	 * @param $endpoint_token
 	 * @param $redirect_uri
+   * @param $state_time_limit time states are valid in seconds
```
Comment (daggerhart) [inline includes/openid-connect-generic-client.php]: Let's align this with the other @params. Might be a tabs vs spaces thing here?

File: includes/openid-connect-generic-client.php [hunk 2]
```diff
@@ -253,7 +255,7 @@ function new_state() {
 	function check_state( $state ) {
 		$states = get_option( 'openid-connect-generic-valid-states', array() );
 		$valid  = FALSE;
-
+
```
Comment (daggerhart) [inline includes/openid-connect-generic-client.php]: Would you mind removing the trailing white space from this empty line?

**Comment by anttileppa** [comment]
Formatting issues should be resolved now

**Comment by daggerhart** [comment]
Awesome, thank you!

**Pull request merged**